File: src/main/java/tk/hiddenname/smarthome/service/hardware/impl/pwm/output/PwmDeviceService.java
package tk.hiddenname.smarthome.service.hardware.impl.pwm.output;

import tk.hiddenname.smarthome.model.signal.PwmSignal;
import tk.hiddenname.smarthome.service.hardware.impl.GPIOService;

public interface PwmDeviceService extends GPIOService {

    PwmSignal getSignal(Long id, boolean reverse);

    PwmSignal setSignal(Long id, boolean reverse, int newSignal);
}

File: src/main/kotlin/tk/hiddenname/smarthome/service/hardware/impl/pwm/output/PwmDeviceService.kt
package tk.hiddenname.smarthome.service.hardware.impl.pwm.output

import tk.hiddenname.smarthome.model.signal.PwmSignal
import tk.hiddenname.smarthome.service.hardware.impl.GPIOService

interface PwmDeviceService : GPIOService {

    fun getSignal(id: Long, reverse: Boolean): PwmSignal

    fun setSignal(id: Long, reverse: Boolean, newSignal: Int): PwmSignal
}